api/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+
+namespace YonderfulApi.Controllers 
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CategoryController : ControllerBase 
+    {
+        private readonly ICategoryService _categoryService;
+        private readonly IPictureService _pictureService;
+        private readonly IMapper _mapper;
+
+        public CategoryController(ICategoryService categoryService, IPictureService pictureService, IMapper mapper)
+        {
+            _categoryService = categoryService;
+            _pictureService = pictureService;
+            _mapper = mapper;
+        }
+
+        [HttpGet("{categoryId}")]
+        public async Task<IActionResult> GetCategory(int categoryId)
+        {
+            var category = await _categoryService.GetCategory(categoryId);
+            if(category == null) {
+                return NotFound();
+            }
+            var categoryDto = transformCategoryDtoForOutput(_mapper.Map<CategoryDto>(category));
+            return Ok(categoryDto);
+        }
+        
+        [HttpGet]
+        public async Task<IActionResult> GetCategoryList() 
+        {
+            var categoryList = await _categoryService.GetCategoryList();
+            if(categoryList == null) {
+                return NotFound();
+            }
+            var categoryDtoList = _mapper.Map<IList<CategoryDto>>(categoryList);
+            return Ok(transformCategoryDtoListForOutput(categoryDtoList));
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> PostCategory(CategoryDto category) 
+        {   
+            var iconId = createPictureFromFileString(category.Icon);
+            var defaultBackgroundId = createPictureFromFileString(category.DefaultBackground);
+            var newCategory = await _categoryService.PostCategory(category.Title, iconId, defaultBackgroundId);
+            if(newCategory == null) {
+                return BadRequest();
+            }
+            return Created(nameof(GetCategory), _mapper.Map<CategoryDto>(newCategory));
+        }
+
+        [HttpDelete]
+        public async Task<ActionResult> DeleteCategory(int categoryId) {
+            var removedCategory = await _categoryService.DeleteCategory(categoryId);
+            return removedCategory ? Ok() : BadRequest();
+        }
+
+        [HttpPut]
+        public async Task<ActionResult> PutCategory(int categoryId, CategoryDto updatedCategory) {
+            var iconId = createPictureFromFileString(updatedCategory.Icon);
+            var defaultBackgroundId = createPictureFromFileString(updatedCategory.DefaultBackground);
+            var newCategory = await _categoryService.PutCategory(categoryId, updatedCategory.Title, iconId, defaultBackgroundId);
+            if(newCategory == null) {
+                return BadRequest();
+            }
+            return Created(nameof(GetCategory), _mapper.Map<CategoryDto>(newCategory));
+        }
+
+        private int createPictureFromFileString(string fileString) {
```

Comment:
I would move all the private methods to the Service to keep the Controller clean (only endpoints actions).
Also, in dotnet, all method names start with an Upercase. Private methods as well.